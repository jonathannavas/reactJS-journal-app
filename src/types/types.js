export const types = {
    
    login: '[auth] login',
    logout: '[logout] logout',

    uiSetError: '[UI] set error',
    uiRemoveError: '[UI] remove error'

}