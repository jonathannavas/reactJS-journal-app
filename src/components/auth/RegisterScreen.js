import React from 'react'
import { Link } from 'react-router-dom'
import { useForm } from '../../hooks/useForm'

export const RegisterScreen = () => {

    const [ formValues, handleInputChange ] = useForm({
        name: 'Gabo',
        email: 'test@email.com',
        password: '12345',
        password2: '12345'
    });

    const { name, email, password, password2 } = formValues;

    const handleRegister = (e) => {
        e.preventDefault();
        console.log(name, email, password, password2);
    }

    return (
        <>
            <h3 className="auth__title"> Register </h3>

            <form onSubmit={ handleRegister }>

                <input 
                    type="text"
                    placeholder="Name"
                    name="name"
                    className="auth__input"
                    autoComplete="off"
                    onChange={ handleInputChange }
                    value={ name }
                />
                <input 
                    type="text"
                    placeholder="Email"
                    name="email"
                    className="auth__input"
                    autoComplete="off"
                    onChange={ handleInputChange }
                    value={ email }
                />
                <input 
                    type="password"
                    placeholder="Password"
                    name="password"
                    className="auth__input"
                    onChange={ handleInputChange }
                    value={ password }
                />
                <input 
                    type="password"
                    placeholder="Confirm password"
                    name="password2"
                    className="auth__input"
                    onChange={ handleInputChange }
                    value={ password2 }
                />
                <button
                    type="submit"
                    className="btn btn-primary btn-block mb-5"
                    onSubmit={ handleRegister }
                    // disabled={true}
                >
                    Register
                </button>

                <Link to="/auth/login" className="link">
                    Already register?
                </Link>

            </form>
        </>
    )
}
