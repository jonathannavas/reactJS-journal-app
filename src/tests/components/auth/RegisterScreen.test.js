import configureStore from 'redux-mock-store';
import thunk from "redux-thunk";
import { mount } from "enzyme";
import { Provider } from "react-redux";

import { RegisterScreen } from "../../../components/auth/RegisterScreen";
import { MemoryRouter } from 'react-router';
import { types } from '../../../types/types';

const middlewares = [thunk];
const mockStore = configureStore(middlewares);
const initState = {
    auth:{},
    ui:{
        loading: false,
        msgError: null
    },
    notes:{
        notes: [],
        active: null
    }
};
let store = mockStore(initState);

jest.mock('../../../actions/auth', () => ({
    startGoogleLogin: jest.fn(),
    startLoginEmailPassword: jest.fn()
}));

// store.dispatch = jest.fn();
const wrapper = mount(
    <Provider store={store}>
        <MemoryRouter>
            <RegisterScreen /> 
        </MemoryRouter>
    </Provider>
);

describe('Tests on RegisterScreen.js', () => {
 


    test('Should make a snapshot', () => {
        expect( wrapper ).toMatchSnapshot();
    });

    test('Should to make a dispatch', () => {
        const emailField = wrapper.find('input[name="email"]');
        emailField.simulate('change', {
            target:{
                value: '',
                name: 'email'
            }
        });

        wrapper.find('form').simulate('submit', {
            preventDefault(){}
        });

        const actions = store.getActions();
        expect(actions[0]).toEqual({
            type: types.uiSetError,
            payload: 'Email not valid'
        })


    });

    test('Should to show an error message', () => {
        const initState = {
            auth:{},
            ui:{
                loading: false,
                msgError: 'Email is incorrect'
            }
        };
        const  store = mockStore(initState);
        const wrapper = mount(
            <Provider store={store}>
                <MemoryRouter>
                    <RegisterScreen /> 
                </MemoryRouter>
            </Provider>
        );


        expect( wrapper.find('.auth__alert-error').exists()).toBe(true);
        expect( wrapper.find('.auth__alert-error').text().trim()).toBe('Email is incorrect');

        
    })

});