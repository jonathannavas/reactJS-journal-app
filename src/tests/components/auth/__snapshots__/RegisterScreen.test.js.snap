// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests on RegisterScreen.js Should make a snapshot 1`] = `
Array [
  <h3
    className="auth__title"
  >
     Register 
  </h3>,
  <form
    className="animate__animated animate__fadeIn animate__faster"
    onSubmit={[Function]}
  >
    <input
      autoComplete="off"
      className="auth__input"
      name="name"
      onChange={[Function]}
      placeholder="Name"
      type="text"
      value="Gabo"
    />
    <input
      autoComplete="off"
      className="auth__input"
      name="email"
      onChange={[Function]}
      placeholder="Email"
      type="text"
      value="test@email.com"
    />
    <input
      className="auth__input"
      name="password"
      onChange={[Function]}
      placeholder="Password"
      type="password"
      value="Multisucredota95."
    />
    <input
      className="auth__input"
      name="password2"
      onChange={[Function]}
      placeholder="Confirm password"
      type="password"
      value="Multisucredota95."
    />
    <button
      className="btn btn-primary btn-block mb-5"
      onSubmit={[Function]}
      type="submit"
    >
      Register
    </button>
    <a
      className="link"
      href="/auth/login"
      onClick={[Function]}
    >
      Already register?
    </a>
  </form>,
]
`;
